src/main/java/dinkplugin/notifiers/DeathNotifier.java

```diff
@@ -168,44 +183,50 @@ private int getKeepCount() {
+            (client.getVarbitValue(Varbits.IN_WILDERNESS) > 0 || WorldUtils.isPvpWorld(client.getWorldType()));
 
         Player localPlayer = client.getLocalPlayer();
+        Predicate<Actor> interacting = a -> INTERACTING.test(localPlayer, a);
 
+        // O(1) fast path based on last outbound interaction
         Actor lastTarget = this.lastTarget.get();
-        if (lastTarget != null && !lastTarget.isDead() && lastTarget.getInteracting() == localPlayer) {
-            if (lastTarget instanceof Player) {
-                Player last = (Player) lastTarget;
-                if (!last.isClanMember() && !last.isFriend() && !last.isFriendsChatMember())
-                    return last;
-            } else if (lastTarget.getCombatLevel() > 0) {
-                return null; // we likely died to this NPC rather than a player
-            }
-        }
+        if (checkLastInteraction(localPlayer, lastTarget, pvpEnabled))
+            return lastTarget;
 
         // find another player interacting with us (that is preferably not a friend or clan member)
-        return client.getPlayers().stream()
-            .filter(other -> other.getInteracting() == localPlayer)
-            .min(
-                Comparator.comparing(Player::isDead) // prefer alive
-                    .thenComparing(Player::isClanMember) // prefer not in clan
-                    .thenComparing(Player::isFriend) // prefer not friend
-                    .thenComparing(Player::isFriendsChatMember) // prefer not fc
-                    .thenComparingInt(p -> Math.abs(localPlayer.getCombatLevel() - p.getCombatLevel())) // prefer similar level
-                    .thenComparing(p -> p.getOverheadIcon() == null) // prefer skulled
-                    .thenComparingInt(p -> -p.getCombatLevel()) // prefer higher level for a given absolute level gap
-            )
-            .orElse(null);
+        if (pvpEnabled) {
+            Optional<Player> pker = client.getPlayers().stream()
+                .filter(interacting)
+                .min(PK_COMPARATOR.apply(localPlayer)); // O(n)
+            if (pker.isPresent())
+                return pker.get();
+        }
+
+        return null;
+    }
+
+    /**
+     * @param localPlayer {@link net.runelite.api.Client#getLocalPlayer()}
+     * @param actor       the {@link Actor} that is a candidate killer from {@link #lastTarget}
+     * @param pvpEnabled  whether a player could be our killer (e.g., in wilderness)
+     * @return whether the specified actor is the likely killer of the local player
+     */
+    private static boolean checkLastInteraction(Player localPlayer, Actor actor, boolean pvpEnabled) {
+        if (!INTERACTING.test(localPlayer, actor))
+            return false;
+
+        if (actor instanceof Player) {
+            Player other = (Player) actor;
+            return pvpEnabled && !other.isClanMember() && !other.isFriend() && !other.isFriendsChatMember();
+        }
+
+        return false;
```

Comment:
Is this meant to return true? or is it intentionally left "wrong" for now as we don't do anything with deaths by NPC actors?

if I read the function correctly, the first if-statement is: "if me and the actor are not interacting, early out & return false"
the second if-statement is: "if the actor is a player, make sure it's actually someone who could pvp with us and return true if that's the case"

Reply:
Correct; this is intentionally "wrong" until the follow-up PR adds logic for NPC checking